ResearchKit/Common/ORKReviewStep.m

```diff
@@ -0,0 +1,98 @@
+ the copyright holders even if such marks are included in this software.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import "ORKReviewStep.h"
+#import "ORKReviewStep_Internal.h"
+#import "ORKHelpers.h"
+#import "ORKReviewStepViewController.h"
+
+
+@implementation ORKReviewStep
+
+- (instancetype)initWithIdentifier:(NSString *)identifier
+                             steps:(nullable NSArray *)steps
+                      resultSource:(nullable id<ORKTaskResultSource, NSSecureCoding>)resultSource {
+    self = [super initWithIdentifier:identifier];
+    if (self) {
+        _steps = [steps copy];
+        _resultSource = resultSource;
+    }
+    return self;
+}
+
++ (instancetype)standaloneReviewStepWithIdentifier:(NSString *)identifier
+                                             steps:(nonnull NSArray *)steps
+                                      resultSource:(nullable id<ORKTaskResultSource, NSSecureCoding>)resultSource {
+    return [[ORKReviewStep alloc] initWithIdentifier:identifier steps:steps resultSource:resultSource];
+}
+
++ (instancetype)embeddedReviewStepWithIdentifier:(nonnull NSString *)identifier {
+    return [[ORKReviewStep alloc] initWithIdentifier:identifier steps:nil resultSource:nil];
+}
+
++ (Class)stepViewControllerClass {
+    return [ORKReviewStepViewController class];
+}
+
+- (nonnull instancetype)initWithCoder:(NSCoder *)aDecoder {
+    self = [super initWithCoder:aDecoder];
+    if (self) {
+        ORK_DECODE_OBJ_CLASS(aDecoder, steps, NSArray);
+        ORK_DECODE_OBJ(aDecoder, resultSource);
+    }
+    return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)aCoder {
+    [super encodeWithCoder:aCoder];
+    ORK_ENCODE_OBJ(aCoder, steps);
+    ORK_ENCODE_OBJ(aCoder, resultSource);
+}
+
+- (BOOL)isEqual:(id)object {
+    __typeof(self) castObject = object;
+    return [super isEqual:object] &&
+    ORKEqualObjects(self.steps, castObject.steps) &&
+    ORKEqualObjects(self.resultSource, castObject.resultSource);
+}
+
+- (instancetype)copyWithZone:(NSZone *)zone {
+    ORKReviewStep *reviewStep = [super copyWithZone:zone];
+    reviewStep->_steps = [self.steps copy];
+    reviewStep->_resultSource = self.resultSource;
+    return reviewStep;
+}
```

Comment:
Do we really need to include `isStandalone`? As a computed property it's just a shorthand for `_steps != nil`. So should`t it be enough to encode/decode/compare and copy the steps array?